clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2285,14 +2313,55 @@ those device functions, such that the kernel attributes are the sum of all
 attributes of all device functions called in this kernel.
 See section 6.7 Attributes for more details.
 
-As Intel extension, ``[[intel::reqd_work_group_size(X, Y, Z)]]`` spelling is allowed
-which features optional arguments `Y` and `Z`, those simplifies its usage if
-only 1- or 2-dimensional ND-range is assumed by a programmer. These arguments
-defaults to ``1``.
+.. code-block:: c++
+
+  [[cl::reqd_work_group_size(4, 4, 4)]] void foo() {}
+
+  class Foo {
+  public:
+    [[cl::reqd_work_group_size(2, 2, 2)]] void operator()() const {}
+  };
+
+  template <int N, int N1, int N2>
+  class Functor {
+  public:
+    [[cl::reqd_work_group_size(N, N1, N2)]] void operator()() const {}
+  };
+
+  template <int N, int N1, int N2>
+  [[cl::reqd_work_group_size(N, N1, N2)]] void func() {}
+
+As Intel extension, ``[[intel::reqd_work_group_size(X, Y, Z)]]`` spelling is
+allowed which features optional arguments `Y` and `Z`, those simplifies its
```

Comment:
How about:
```
As an Intel extension, the [[intel::reqd_work_group_size(X, Y, Z)]] spelling is supported. This spelling allows the Y and Z arguments to be optional. If not provided by the user, the value of Y and Z defaults to 1. This simplifies usage of the attribute when a 1- or 2-dimensional ND-range is assumed.
```

Reply:
You should have an example that elides the Y and Z arguments, like:
```
[[intel::reqd_work_group_size(5)]] // identical to [[intel::reqd_work_group_size(5, 1, 1)]]
void quux() {}
```

Reply:
Done